### PR TITLE
Adding transition config cron

### PR DESCRIPTION
**Pull request by nsabri1**
This adds a new cron which runs every Monday to Friday every hour between 9am - 7pm.

The cron in summary clones https://github.com/alphagov/transition-config to a NFS share mounted to the directory transition-config which is then mounted to /app/data/transition-config where the rake command is run **import:all:orgs_sites_hosts**.

The mount point is also mounted to Transition app  **/app/data/transition-config**.

The location of the mount is what is used in the current EC2 variant.

Did consider using a shared PVC but this would have added unneeded complexity and opted for NFS which is already been used for shared storage for other apps.

I also considered using the existing Cron - but felt as this is quite bespoke and therefore created a new cron for it specifically.

Commits contain more information. 

This has been tested in integration.
https://trello.com/c/eznEJ8i0/1120-check-transition-load-site-config-job-works-on-eks

### CODE REVIEW

**Review by sengi — COMMENTED** [review]
Nice. I think NFS is a good call here, given our existing usage of it and it saves a ton of other hassle.

(One day we could make it use S3 or a shared readOnly volume or something but that'd be a distraction right now.)

File: charts/generic-govuk-app/templates/transition-config-cron.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+{{- if .Values.transitionConfigEnabled -}}
```
[the remark this answers is not in the document]
Comment (sengi) [reply charts/generic-govuk-app/templates/transition-config-cron.yaml:1]: Like we do with whitehall-admin.

[the remark this answers is not in the document]
Comment (nsabri1) [reply charts/generic-govuk-app/templates/transition-config-cron.yaml:1]: I have tried this but issue i face is the container which require Githib cli - to run git clone of the repo transition-config

[the remark this answers is not in the document]
Comment (sengi) [reply charts/generic-govuk-app/templates/transition-config-cron.yaml:1]: Oh, I see. In that case, could either keep this and remove the bits you don't need (and make `clone-transition-config-repo` an `initContainer` so it doesn't race with `update-transition-config`), or pass the initcontainer spec into `cronTasks` and use the existing template.

File: charts/generic-govuk-app/templates/transition-config-cron.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ $fullName }}-transition-config-cron
+  labels:
+    {{- include "generic-govuk-app.labels" $ | nindent 4 }}
+    app: {{ $fullName }}-transition-config-cron
+    app.kubernetes.io/name: {{ $fullName }}-transition-config-cron
+    app.kubernetes.io/component: transition-config-cron
```
Comment (sengi) [inline charts/generic-govuk-app/templates/transition-config-cron.yaml:12]: Similar idea here:
```suggestion
    app.kubernetes.io/component: config-cron
```
(and similarly below)

File: charts/generic-govuk-app/templates/transition-config-cron.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            - name: tmp
+              emptyDir: {}
+            {{- with $.Values.extraVolumes }}
+              {{- . | toYaml | trim | nindent 12 }}
+            {{- end }}
+          initContainers:
+            - name: clone-transition-config-repo
+              image: 172025368201.dkr.ecr.eu-west-1.amazonaws.com/github-cli:latest
+              command: ["/bin/bash"]
+              args: ["-c", "git clone https://github.com/alphagov/transition-config /transition-config/"]
```
Comment (sengi) [inline charts/generic-govuk-app/templates/transition-config-cron.yaml:48]: It doesn't look like you need bash here. Any reason you can't just run git directly?
```suggestion
              command: ["git"]
              args:
                - clone
                - --depth 1
                - https://github.com/alphagov/transition-config
                - /transition-config/
```
(Can also add `--depth 1` to save time/resources.)

**Review by sengi — APPROVED** [review]

File: charts/generic-govuk-app/templates/transition-config-cron.yaml (hunk 3 above)
Comment (nsabri1) [reply charts/generic-govuk-app/templates/transition-config-cron.yaml:48]: this is a good call - let me test this

**Pull request merged**